lib/iris/coords.py

```diff
@@ -909,14 +917,25 @@ def is_compatible(self, other, ignore=None):
         return compatible
 
     @property
-    def dtype(self):
+    def points_dtype(self):
```

Comment:
So we'd have `coord.dtype` and `coord.bounds_dtype`, or `coord.dtype`, `coord.points_dtype` and `coord.bounds_dtype`? My preference is probably for only having two dtype attributes as `coord.dtype` and `coord.points_dtype` would surely always give the same answer...